Sources/LiveKit/Publications/RemoteTrackPublication.swift

```diff
@@ -164,22 +165,26 @@ public class RemoteTrackPublication: TrackPublication {
 
 private extension RemoteTrackPublication {
 
-    func checkCanModifyTrackSettings() -> Promise<Void> {
+    var isAdaptiveStreamEnabled: Bool { (participant?.room.options ?? RoomOptions()).adaptiveStream && .video == kind }
 
-        Promise<Void>(on: .sdk) { () -> Void in
+    var engineConnectionState: ConnectionState {
 
-            guard let adaptiveStream = self.participant?.room.options.adaptiveStream else {
-                throw TrackError.state(message: "The state of AdaptiveStream is unknown")
-            }
+        guard let participant = participant else {
+            log("Participant is nil", .warning)
+            return .disconnected()
+        }
 
-            guard !adaptiveStream else {
-                throw TrackError.state(message: "Cannot modify this track setting when using AdaptiveStream")
-            }
+        return participant.room.engine._state.connectionState
+    }
 
-            guard self.subscribed else {
-                throw TrackError.state(message: "Cannot modify this track setting when not subscribed")
-            }
-        }
+    var userCanModifyTrackSettings: Bool {
+
+        // adaptiveStream must be disabled and must be subscribed
+        guard let adaptiveStream = self.participant?.room.options.adaptiveStream,
```

Comment:
should this be `!isAdaptiveStreamEnabled` ? it's still possible to modify audio tracks (disabling them).

Reply:
Got it, audio should be ok to modify.